core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1255 @@
+  });
+
+  beforeEach(() => {
+    alertsService = TestBed.inject(AlertsService);
+    answerClassificationService = TestBed.inject(AnswerClassificationService);
+    contextService = TestBed.inject(ContextService);
+    expressionInterpolationService =
+      TestBed.inject(ExpressionInterpolationService);
+    questionObjectFactory = TestBed.inject(QuestionObjectFactory);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    questionPlayerEngineService = TestBed.inject(QuestionPlayerEngineService);
+    urlService = TestBed.inject(UrlService);
+  });
+
+  it('should load questions when initialized', () => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+
+    let questionPlayerSpy =
+      spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    let explorationIdSpy = spyOn(contextService, 'getExplorationId')
+      .and.returnValue('explorationId1');
+    let questionPlayerModeSpy =
+      spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(true);
+    let versionSpy =
+      spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    expect(questionPlayerSpy).toHaveBeenCalled();
+    expect(explorationIdSpy).toHaveBeenCalled();
+    expect(questionPlayerModeSpy).toHaveBeenCalled();
+    expect(versionSpy).toHaveBeenCalled();
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+    expect(questionPlayerEngineService.getExplorationId())
+      .toBe('explorationId1');
+    expect(questionPlayerEngineService.questionPlayerMode).toBe(true);
+    expect(questionPlayerEngineService.getExplorationVersion()).toBe(1);
+    expect(questionPlayerEngineService.questions.length).toBe(3);
+  });
+
+  it('should update the exploration version if it' +
+    'is not in question player mode when initialized', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = sampleExplorationDict;
+    exploration.version = 2;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('explorationId1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+
+    let readOnlySpy =
+      spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+        .and.callFake((expID, version) => {
+          expect(version).toBe(1);
+          return Promise.resolve(exploration);
+        });
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    tick();
+
+    expect(questionPlayerEngineService.questionPlayerMode).toBe(false);
```

Comment:
Why is this a post-condition check? The test description suggests that making this hold is part of the setup for the test.

Reply:
removed.